pkg/webhook/cspc_operations.go

```diff
@@ -509,3 +470,203 @@ func GetNewBDFromRaidGroups(newRG, oldRG *cstor.RaidGroup) map[string]string {
 	newToOldBlockDeviceMap[newBD] = oldBD
 	return newToOldBlockDeviceMap
 }
+
+// raidGroups contains list of oldraid groups and newraid groups
+type raidGroups struct {
+	oldRaidGroups []cstor.RaidGroup
+	newRaidGroups []cstor.RaidGroup
+	rgType        string
+}
+
+func getNewBDsFromStripeSpec(oldRg, newRg cstor.RaidGroup) []string {
+	mapOldRg := map[string]bool{}
+	bds := []string{}
+	for _, bd := range oldRg.CStorPoolInstanceBlockDevices {
+		mapOldRg[bd.BlockDeviceName] = true
+	}
+	for _, bd := range newRg.CStorPoolInstanceBlockDevices {
+		if !mapOldRg[bd.BlockDeviceName] {
+			bds = append(bds, bd.BlockDeviceName)
+		}
+	}
+	return bds
+}
+
+// getBDsFromRaidGroups return list of blockdevices from list of raid groups
+func getBDsFromRaidGroups(rgs []cstor.RaidGroup) []string {
+	bds := []string{}
+	for _, rg := range rgs {
+		for _, bd := range rg.CStorPoolInstanceBlockDevices {
+			bds = append(bds, bd.BlockDeviceName)
+		}
+	}
+	return bds
+}
+
+// getExpandedRaidGroups returns the raid groups only if pool expansion is done
+// by the users
+func getExpandedRaidGroups(
```

Comment:
This function logic should be changed according  we have check-in writecache group also